docs/AuthenticationMethods.md

```diff
@@ -0,0 +1,107 @@
+#Authentication Methods
```

Comment:
Consider adding instead a separate "Authentication.md" file, with Keeping Authentication Parameters in Key Vault as a section, and modifying Sink/Source to link to this document. 
Regardless, change the name to start with a capital letter as the rest of the docs